docker/src/test/java/org/jclouds/docker/features/NetworkApiLiveTest.java

```diff
@@ -0,0 +1,123 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.docker.features;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertNull;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.jclouds.docker.compute.BaseDockerApiLiveTest;
+import org.jclouds.docker.domain.Config;
+import org.jclouds.docker.domain.Container;
+import org.jclouds.docker.domain.Image;
+import org.jclouds.docker.domain.Network;
+import org.jclouds.docker.options.CreateImageOptions;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.google.common.base.Predicates;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Iterables;
+
+@Test(groups = "live", testName = "NetworkApiLiveTest", singleThreaded = true)
+public class NetworkApiLiveTest extends BaseDockerApiLiveTest {
+
+   private static final String NETWORK_NAME = "JCLOUDS_NETWORK";
+
+   private Network network = null;
+   protected Image image = null;
+   private Container container;
+
+   @BeforeClass
+   protected void init() {
+
+      if (api.getImageApi().inspectImage(ALPINE_IMAGE_TAG) == null) {
+         CreateImageOptions options = CreateImageOptions.Builder.fromImage(ALPINE_IMAGE_TAG);
+         InputStream createImageStream = api.getImageApi().createImage(options);
+         consumeStream(createImageStream);
+      }
+      image = api.getImageApi().inspectImage(ALPINE_IMAGE_TAG);
+      assertNotNull(image);
+
+      Config containerConfig = Config.builder().image(image.id())
+              .cmd(ImmutableList.of("sh", "-c", "touch hello; while true; do echo hello world; sleep 1; done"))
+              .build();
+      container = api.getContainerApi().createContainer("jclouds-test-network", containerConfig);
+      api.getContainerApi().startContainer(container.id());
+      container = api.getContainerApi().inspectContainer(container.id());
+   }
+
+   @AfterClass
+   protected void tearDown() {
+      if (container != null) {
+         api.getContainerApi().stopContainer(container.id());
+         api.getContainerApi().removeContainer(container.id());
+      }
+      if (network != null) {
+         api().removeNetwork(network.id());
+      }
+   }
+
+   public void testCreateNetwork() throws IOException, InterruptedException {
+      network = api().createNetwork(Network.create(NETWORK_NAME, null, null, null, null, ImmutableMap.<String, Network.Details> of(), ImmutableMap.<String, String> of()));
```

Comment:
Not sure, maybe when I'll add auto value builders things will be easier?
I think there are a bunch of options:
- multiple Network.create() 
- `NetworkCreate` 
- Network.builder(). ... .build() using AutoValue builders
but maybe we should have some sort of guidelines.
wdyt?

Happy to solve it here or in a subsequent PR.